packages/skeleton-svelte/src/routes/components/ratings/+page.svelte

```diff
@@ -1,163 +1,122 @@
 <script lang="ts">
-	import Rating from '$lib/components/Rating/Rating.svelte';
-	import Star from 'lucide-svelte/icons/star';
-	import Skull from 'lucide-svelte/icons/skull';
-	import Bone from 'lucide-svelte/icons/bone';
+	import { Rating } from '$lib/index.js';
+	import { Star, StarHalf } from 'lucide-svelte';
 
-	let interactiveValue = $state(3);
+	let value = $state(3);
 </script>
 
 <div class="space-y-10">
 	<header>
-		<h1 class="h1">Ratings</h1>
+		<h1 class="h1">Rating</h1>
 	</header>
-
 	<section class="space-y-4">
-		<div class="flex items-center gap-8">
-			<div class="w-48">
-				<Rating value={5}>
-					{#snippet iconFull()}
-						<Star size={24} class="fill-surface-950-50" />
-					{/snippet}
+		<h2 class="h2">Default</h2>
+		<p>{value}</p>
+		<Rating bind:value>
+			{#each { length: 5 } as _, index}
+				<Rating.Item {index}>
 					{#snippet iconEmpty()}
 						<Star size={24} />
 					{/snippet}
-				</Rating>
-			</div>
-			<h3 class="h3">5.0</h3>
-		</div>
-		<div class="flex items-center gap-8">
-			<div class="w-48">
-				<Rating value={0}>
-					{#snippet iconFull()}
-						<Star size={24} class="fill-surface-950-50" />
+					{#snippet iconHalf()}
+						<StarHalf size={24} class="fill-surface-950-50" />
```

Comment:
You did a great job with the half stars, but I see the SVG fill isn't quite cutting it here for Lucide. We need a full outline and half fill. Sadly Lucide doesn't seem to a means to do that. [Maybe we can request that](https://github.com/lucide-icons/lucide/issues/new?assignees=&labels=%F0%9F%99%8C+icon+request&projects=&template=01_icon_request.yml)? Seems like an easy add.

Unless you can find some SVG trickery to only fill half of outlined star, we may need to implement our own custom SVGs here.

On that note, we did have a request on Discord to provide some generic star SVGs as built-in placeholders. Maybe we can find a generic enough set of icons to fulfill this purpose. I'm thinking two SVGs:

1. Full star - can customize the stroke/fill; can be used for full/empty states
2. Half Star - full star in shape for stroke, but fill only ever covers half

That should take care of all three states: full/empty/half. See me if I can clarify this at all!

Reply:
Mock:

![Screenshot 2024-08-05 at 3 20 31 PM](https://github.com/user-attachments/assets/bb92daf5-dc5b-4935-a77f-b30b85be676c)

Reply:
I've done 3 SVG's for now because I simply couldn't get it working how I wanted. I've done the rest afaik. You may review and change if needed.

Reply:
Looks good. I'll generate new artwork for these asap.